apps/user/v1/views.py

```diff
@@ -17,4 +25,36 @@ def post(self, request):
             user = serializer.validated_data
             return Response({"message": "Successful login"}, status=status.HTTP_200_OK)
         else:
-            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
\ No newline at end of file
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        
+class PasswordResetView(APIView):
+    def post(self, request):
+        emails = EmailSender()
+        serializer = PasswordResetSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        username_or_email = serializer.validated_data['username_or_email']
+        user = User.objects.filter(Q(email=username_or_email) | Q(username=username_or_email)).first()
+        if not user:
+            raise Http404("No user with this username/email address has been found.")
+
+        private_key = read_private_key()
+
+        payload = {
+            'user_id': user.id,
+            'exp': datetime.now(timezone.utc) + timedelta(hours=1),
+            'iat': datetime.now(timezone.utc)
+        }
+
+        token = jwt.encode(payload, private_key, algorithm='RS256')
+        
+        frontend_url = os.getenv('FRONTEND_URL')
+
+        reset_link = f"{frontend_url}/reset-password/?k={token}"
+        
+        emails.send_email_html(
+            user.email,
+            "Password Reset",
+            f"Hello {user.username},<br><br>Click <a href='{reset_link}'>here</a> to reset your password."
```

Comment:
Esto sería el cuerpo del email no? No sería mejor dejar la llamada al .html directamente y que cuando se haga el merge con la plantilla ya se vea bien?

Es decir, me refiero a crear un .html donde tenga que estar con el texto html que esta aqui. Luego @crist6494 al hacer la plantilla crearia un archivo en el mismo sitio y lugar con su codigo. Al hacer merge de las dos se quedaría el codigo de cristian con la llamada de @Atrujillo02 en este fichero.

Reply:
Como prefiráis

Reply:
como queráis hacerlo sino se puede quedar asi cuando Cristian haga el cuerpo html, ya cambio yo esta parte o quedo con el y le digo donde tenemos que ponerlo desde su rama.

Reply:
Lo digo más que nada para que directamente lo tenga bien puesto pa no hacer una tarea diferente pa cambiar eso. Si se puede hacer del tirón y está en su scope mejor